pandas/tests/computation/test_eval.py

```diff
@@ -737,6 +737,13 @@ def test_and_logic_string_match(self):
         assert pd.eval(f"{event.str.match('hello').a}")
         assert pd.eval(f"{event.str.match('hello').a and event.str.match('hello').a}")
 
+    def test_using_numpy(self):
+        # GH 58041
+        df = Series([0.2, 1.5, 2.8], name="a").to_frame()
+        res = df.eval("@np.floor(a)")
+        expected = np.floor(df["a"])
+        tm.assert_series_equal(expected, res, check_names=False)
```

Comment:
Can you set the proper name on `expected` and then remove `check_names=False`.

Reply:
Unfortunately, I can't. The issue is that `df.eval(@np.floor(a))` will have different `name`s in Linux unittests compared to Win/MacOS unittest. Meaning if I just remove `check_names` and:
* leave as-is, the Linux unittests will fail
* change the name of `expected` to `"a"`, the Win & MacOS unittests will fail
Please see my previous comment and/or #58069 for a little bit more details.

Reply:
On a further thought, we can sort of work around the issue by explicitly stating the `engine`, which is arguably something we should do anyway. I've updated the PR accordingly -- and also moved the test into more fitting test file. Please take a look and let me know what you think.